### ISSUE TITLE
Cannot use any new application without manual intervention in the database

### ISSUE THREAD

**Issue by tzok** [indigo-dc/fgAPIServer#34]
When you add a new application using REST API, you still need to manually dig in the database to use it. This is because by default every application has zero groups granted any access... 

This should be fixed in a way that a user who is mapped during application creation has all access to it. And maybe users from group "admin" also has all access rights while group "users" has only view & execute permissions.

**Comment by ricsxn** [comment]
That’s fine, do you mind to open an issue on this, please?

Regards,
R.

**Comment by ricsxn** [comment]
This will be available in next release by the code lines in fgapiberverdb.py:
                # Enable the groups owned by the installing user to
                # execute the app
                fgapisrv_db.enable_app_by_userid(
                    user_id,
                    app_id)

**Comment by tzok** [comment]
Unfortunately the error is still there. When I add a new application using POST to `/apis/v1.0/applications`, then there are no entries in the `fg_group_apps` table. In consequence, nobody is able to use the newly installed application without manual intervention into the database.

**Comment by ricsxn** [comment]
In fgaiserverdb.py the function enable_app_by_userid is in charge of this

It is called by fgapiserver.py with:

fgapisrv_db.enable_app_by_userid(
                    user_id,
                    app_id)

It executes:

select group_id from fg_user_group where user_id = %s

"insert into fg_group_apps (group_id,app_id, creation)"
                       " values (%s,%s,now())"

I should understand what went wrong in your case especially regardless the user_id.

Regards,
Riccardo

**Comment by tzok** [comment]
I am using a script like that:
```bash
curl -H "Authorization: Bearer $1" \
    -H 'Content-Type: application/json' \
    -X POST -d "$(cat application.json)" \
    https://fgw01.ncg.ingrid.pt/apis/v1.0/applications
```

It worked fine and I have created applications with id `3` and `4` on fgw01.ncg.ingrid.pt. But the `fg_group_apps` table was not populated with data. There are no entries for `3` or `4` there. Well, right now there are some entries on fgw01 because I have added them, but before I did this, there were only entries for ids `0`, `1` and `2`)
